core/src/main/java/jenkins/slaves/EncryptedSlaveAgentJnlpFile.java

```diff
@@ -35,27 +36,23 @@
  */
 public class EncryptedSlaveAgentJnlpFile implements HttpResponse {
     /**
-     * The object that owns the Jelly view that renders JNLP file.
-     * For example {@link SlaveComputer}.
+     * The object that owns the Jelly view that renders JNLP file this is always 
+     * some type of {@link SlaveComputer}.
```

Comment:
This is unfortunately an incorrect assumption.  The only reason you need to have changed this is to get the MAC of the JNLP slave, but you can achieve this by keeping binary compatibility.